crates/accelerate/src/pauli_lindblad_map/pauli_lindblad_map_class.rs

```diff
@@ -0,0 +1,1734 @@
+    ///
+    /// Args:
+    ///     indices (Sequence[int]): The indices for which Paulis must be kept.
+    ///
+    /// Returns:
+    ///     A new Pauli Lindblad map where every Pauli on the given `indices` has been kept and all other
+    ///     Paulis have been dropped.
+    ///
+    /// Examples:
+    ///
+    ///     .. code-block:: python
+    ///
+    ///         >>> pauli_map_in = PauliLindbladMap.from_list([("XXIZI", 2.0), ("IIIYZ", 0.5), ("ZIIXY", -0.25)])
+    ///         >>> pauli_map_out = pauli_map_in.keep_paulis([1, 2, 4])
+    ///         >>> assert pauli_map_out == PauliLindbladMap.from_list([("XIIZI", 2.0), ("IIIYI", 0.5), ("ZIIXI", -0.25)])
+    #[pyo3(signature = (/, indices))]
+    pub fn keep_paulis(&self, indices: Vec<u32>) -> PyResult<Self> {
+        let inner = self.inner.read().map_err(|_| InnerReadError)?;
+
+        let max_index = match indices.iter().max() {
+            Some(&index) => index,
+            None => 0,
+        };
+        if max_index >= inner.num_qubits() {
+            let num_qubits = inner.num_qubits();
+            return Err(PyValueError::new_err(format!(
+                "cannot keep Paulis for index {max_index} in a {num_qubits}-qubit PauliLindbladMap"
+            )));
+        }
+
+        Ok(inner
+            .drop_paulis(
+                (0..self.num_qubits()?)
+                    .filter(|index| !indices.contains(index))
+                    .collect(),
+            )?
+            .into())
+    }
+
+    /// Sample sign and Pauli operator pairs from the map. Each sign is represented by a boolean,
+    /// with `True` representing `+1`, and False representing `-1`.
+    ///
+    /// Given the quasi-probability representation given in the class level documentation, each
+    /// sample is drawn via the following process:
+    ///     * Initialize the sign boolean, and a :class`~.QubitSparsePauli` instance to the identity
+    ///     operator.
+    ///     * Iterate through each Pauli in the map. Using the pseudo-probability associated with
+    ///     each operator, randomly choose between applying the operator or not.
+    ///     * If the operator is applied, update the :class`QubitSparsePauli` by multiplying it with
+    ///     the Pauli. If the rate associated with the Pauli is negative, flip the sign boolean.
+    ///
+    /// The results are returned as a 1d array of booleans, and the corresponding sampled qubit
+    /// sparse paulis in the form of a :class:`~.QubitSparsePauliList`.
+    ///
+    /// Args:
+    ///     n_samples (int): Number of samples to draw.
+    ///     seed (int): Random seed.
+    /// Returns:
+    ///     signs, qubit_sparse_pauli_list: The boolean array of signs and the list of qubit sparse
+    ///     paulis.
+    #[pyo3(signature = (n_samples, seed))]
+    pub fn sample<'py>(
+        &self,
+        py: Python<'py>,
+        n_samples: u64,
+        seed: Option<u64>,
+    ) -> PyResult<Bound<'py, PyTuple>> {
```

Comment:
Btw, while making my other comment, i noticed that the sample method, from python, doesn't seem to recognize that the seed is optional. I get `TypeError: PauliLindbladMap.sample() missing 1 required positional argument: 'seed'` if only specifying the size.

Reply:
Dan: you need to add a signature annotation to tell PyO3 about Python-space defaults, if you want them to exist.  Something like this:
```rust
#[pyo3(signature = (n_samples, *, seed=None)]
pub fn sample<'py>(...)
```